qos-service-impl/src/main/java/com/palantir/atlasdb/qos/client/AtlasDbQosClient.java

```diff
@@ -98,4 +98,12 @@ public static AtlasDbQosClient create(QosRateLimiters rateLimiters) {
         }
     }
 
+    @Override
+    public void close() {
+        try {
+            rateLimiters.read().close();
+        } finally {
+            rateLimiters.write().close();
+        }
+    }
```

Comment:
This is clever, I didn't think of this 😄 

Might not scale if there were more resources, but since there were just two...

Reply:
It messes up the exceptions — if an exception is thrown in the first block and in the finally block, the finally one just shadows the first one. For a correct behavior we'd need to add the first one to the suppressed exceptions of the second one... but I think it should be fine.